pkg/transformers/json/transformer.go

```diff
@@ -14,21 +14,34 @@ import (
 
 const sep = "/"
 
-var keys = [...]string{"publisher", "protocol", "channel", "subtopic"}
-
 var (
+	keys = [...]string{"publisher", "protocol", "channel", "subtopic"}
+
 	// ErrTransform represents an error during parsing message.
-	ErrTransform         = errors.New("unable to parse JSON object")
-	ErrInvalidKey        = errors.New("invalid object key")
+	ErrTransform = errors.New("unable to parse JSON object")
+	// ErrInvalidKey represents the use of a reserved message field.
+	ErrInvalidKey = errors.New("invalid object key")
+
 	errUnknownFormat     = errors.New("unknown format of JSON message")
 	errInvalidFormat     = errors.New("invalid JSON object")
 	errInvalidNestedJSON = errors.New("invalid nested JSON object")
+
+	timestamps = []Timestamps{}
 )
 
 type funcTransformer func(messaging.Message) (interface{}, error)
 
+// Timestamps represents the message fields to use as timestamp
+type Timestamps struct {
+	FieldName   string `toml:"field_name"`
+	FieldFormat string `toml:"field_format"`
+	Location    string `toml:"location"`
+}
+
 // New returns a new JSON transformer.
-func New() transformers.Transformer {
+func New(ts []Timestamps) transformers.Transformer {
+	// TODO: Improve the timestamp config
+	timestamps = ts
```

Comment:
Attach timestamps from config to transformer struct (that is, replace `funcTransformer` with a struct that holds its config).